lib/services/ios-project-service.ts

```diff
@@ -194,365 +191,29 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
-<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-<plist version="1.0">
-<dict>
-    <key>method</key>
-	<string>${exportOptionsMethod}</string>`;
-		if (options && options.provision) {
-			plistTemplate += `    <key>provisioningProfiles</key>
-<dict>
-	<key>${projectData.projectIdentifiers.ios}</key>
-	<string>${options.provision}</string>
-</dict>`;
-		}
-		plistTemplate += `
-    <key>uploadBitcode</key>
-    <false/>
-    <key>compileBitcode</key>
-    <false/>
-</dict>
-</plist>`;
-
-		// Save the options...
-		temp.track();
-		const exportOptionsPlist = temp.path({ prefix: "export-", suffix: ".plist" });
-		this.$fs.writeFile(exportOptionsPlist, plistTemplate);
-
-		// The xcodebuild exportPath expects directory and writes the <project-name>.ipa at that directory.
-		const exportPath = path.resolve(path.dirname(archivePath));
-		const exportFile = path.join(exportPath, projectData.projectName + ".ipa");
-
-		await this.xcodebuild(
-			[
-				"-exportArchive",
-				"-archivePath", archivePath,
-				"-exportPath", exportPath,
-				"-exportOptionsPlist", exportOptionsPlist
-			],
-			projectRoot, buildConfig.buildOutputStdio);
-		return exportFile;
-	}
-
-	private xcbuildProjectArgs(projectRoot: string, projectData: IProjectData, product?: "scheme" | "target"): string[] {
-		const xcworkspacePath = path.join(projectRoot, projectData.projectName + ".xcworkspace");
-		if (this.$fs.exists(xcworkspacePath)) {
-			return ["-workspace", xcworkspacePath, product ? "-" + product : "-scheme", projectData.projectName];
-		} else {
-			const xcodeprojPath = path.join(projectRoot, projectData.projectName + ".xcodeproj");
-			return ["-project", xcodeprojPath, product ? "-" + product : "-target", projectData.projectName];
-		}
-	}
-
 	public async buildProject(projectRoot: string, projectData: IProjectData, buildConfig: IBuildConfig): Promise<void> {
-		const basicArgs = [
-			'SHARED_PRECOMPS_DIR=' + path.join(projectRoot, 'build', 'sharedpch')
-		];
-
+		const platformData = this.getPlatformData(projectData);
 		const handler = (data: any) => {
 			this.emit(constants.BUILD_OUTPUT_EVENT_NAME, data);
 		};
 
 		if (buildConfig.buildForDevice) {
+			await this.$iOSSigningService.setupSigningForDevice(projectRoot, projectData, buildConfig);
 			await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
 				this.$childProcess,
 				handler,
-				this.buildForDevice(projectRoot, basicArgs, buildConfig, projectData));
+				this.$xcodebuildService.buildForDevice(platformData, projectData, buildConfig));
 		} else {
 			await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
 				this.$childProcess,
 				handler,
-				this.buildForSimulator(projectRoot, basicArgs, projectData, buildConfig));
+				this.$xcodebuildService.buildForSimulator(platformData, projectData, buildConfig));
 		}
 
+		// TODO: Check if we need to validate the identifier here
```

Comment:
Did you check that?

Reply:
Yes, I checked it -> we need this logic.